### PR TITLE
Updated CHANGELOG for 2.35.0

### PR DESCRIPTION
**Pull request by ygong**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
+# Release 2.35.0 22-03-2022
+
+- Update Envoy to v1.21.1 and Bazel to v5.0.0 (#670)
+- Remove envoy runtime flag preserve_downstream_scheme (#667)
+- Add option to allow discovery APIs (#665)
```
Comment (nareddyt) [inline CHANGELOG.md:5]: You can remove this one, it is not meant for 3P

Comment (nareddyt) [reply CHANGELOG.md:5]: Correct

Comment (ygong) [reply CHANGELOG.md]: Done

**Review by nareddyt — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,11 @@
+# Release 2.35.0 22-03-2022
+
+- Update Envoy to v1.21.1 and Bazel to v5.0.0 (#670)
+- Remove envoy runtime flag preserve_downstream_scheme (#667)
+- Correctly escape user-provided regex paths (#664)
+- Expose the `--config-yaml` envoy flag  (#662)
+
+===============================================
```
Comment (TAOXUY) [inline CHANGELOG.md:8]: Remove line8

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,11 @@
+# Release 2.35.0 22-03-2022
+
+- Update Envoy to v1.21.1 and Bazel to v5.0.0 (#670)
```
Comment (TAOXUY) [inline CHANGELOG.md:3]: NIT: only "- Update Envoy to v1.21.1 (#670)" for line 3

File: CHANGELOG.md (hunk 2 above)
Comment (ygong) [reply CHANGELOG.md]: Done

File: CHANGELOG.md (hunk 3 above)
Comment (ygong) [reply CHANGELOG.md]: Done.

**Review by qiwzhang — APPROVED** [review]

**Review by TAOXUY — APPROVED** [review]

**Pull request merged**